app/controllers/posts_controller.rb

```diff
@@ -27,6 +27,16 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    user = User.find(params[:user_id])
+    user.posts_counter -= 1
+    @post.destroy!
+    user.save
+    flash[:success] = 'Post deleted!'
+    redirect_to user_posts_path(user.id)
+  end
+
   private
 
```

Comment:
- [ ] The Post Controller uses **CanCanCan** role-based authorization hence, you will require to do something of this sort:

```ruby
class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```